clusterloader2/pkg/test/simple_test_executor.go

```diff
@@ -55,17 +56,50 @@ func (ste *simpleTestExecutor) ExecuteTest(ctx Context, conf *api.Config) *error
 	klog.Infof("AutomanagedNamespacePrefix: %s", ctx.GetClusterFramework().GetAutomanagedNamespacePrefix())
 	defer cleanupResources(ctx)
 	ctx.GetTuningSetFactory().Init(conf.TuningSets)
+	chaosMonkeyWaitGroup, errList := ste.ExecuteTestSteps(ctx, conf)
+
+	if chaosMonkeyWaitGroup != nil {
+		// Wait for the Chaos Monkey subroutine to end
+		klog.Info("Waiting for the chaos monkey subroutine to end...")
+		chaosMonkeyWaitGroup.Wait()
+		klog.Info("Chaos monkey ended.")
+	}
+
+	for _, summary := range ctx.GetMeasurementManager().GetSummaries() {
+		if ctx.GetClusterLoaderConfig().ReportDir == "" {
+			klog.Infof("%v: %v", summary.SummaryName(), summary.SummaryContent())
+		} else {
+			testDistinctor := ""
+			if ctx.GetClusterLoaderConfig().TestScenario.Identifier != "" {
+				testDistinctor = "_" + ctx.GetClusterLoaderConfig().TestScenario.Identifier
+			}
+			// TODO(krzysied): Remember to keep original filename style for backward compatibility.
+			fileName := strings.Join([]string{summary.SummaryName(), conf.Name + testDistinctor, summary.SummaryTime().Format(time.RFC3339)}, "_")
+			filePath := path.Join(ctx.GetClusterLoaderConfig().ReportDir, strings.Join([]string{fileName, summary.SummaryExt()}, "."))
+			if err := ioutil.WriteFile(filePath, []byte(summary.SummaryContent()), 0644); err != nil {
+				errList.Append(fmt.Errorf("writing to file %v error: %v", filePath, err))
+				continue
+			}
+		}
+	}
+	klog.Infof(ctx.GetChaosMonkey().Summary())
+	return errList
+}
+
+// ExecuteTestSteps executes all test steps provided in configuration
+func (ste *simpleTestExecutor) ExecuteTestSteps(ctx Context, conf *api.Config) (*sync.WaitGroup, *errors.ErrorList) {
```

Comment:
I like the overall approach, but that it is introducing unnecessary complication.

Let's:
- create stopCh and ChaosMonkey outside of this function (in the ExecuteTest one)
- this one then doesn't need to know (and return) about WaitGroup

Reply:
No problem, I refactored it to only return the errorList